### ISSUE TITLE
Optimize development tasks for working with multiple packages

### ISSUE THREAD

**Issue by sam-b-rose** [Shopify/polaris-react-archive#5666]
## Overview

**Vault project: TBD**

> The goal – make contributing to Polaris easy! 👟 ⚽ 🥅

We’ve cleaned up linting configurations and code formatting, but we still lack consistency with TypeScript configs and bundler setups. Some packages are using Loom, some use tsc (the TypeScript compiler), and others use Rollup directly. We can improve how our root-level and package scripts are run to be smarter (only run when needed, cached ouput) and in parallel. We can also standardize what scripts should be part of every package: build, dev, lint, test, and clean to name a few essentials.

These improvements will make it more clear when developing in Polaris what scripts are available, how they interact with CI, and what to include when adding a new package to the monorepo. It will also improve CI and local development speeds by only running scripts when relevant files are changed, paralize tasks, and cache build outputs.

### Tasks

- [x] #5681
- [x] #5680
- [x] #5682
- [x] #5753
- [x] #5755
- [x] #5756
- [x] #5814

Optional stretch goals

- [x] #5679
- [x] #5678

## Outcomes

- Standardized TypeScript configurations for easier TypeScript management, type checking, and clarity around TypeScript configuration options/presets
- Simple standardized set of package scripts for individual and root usage
- Parallel and cached running of scripts for CI and local development using Turborepo (or other such as [Google/wireit](https://github.com/google/wireit))
- Improve working across packages (ex: local updates to tokens should cascade to an update in the website)

**Comment by sam-b-rose** [comment]
Going to close out this issue as Done. The remaining stretch goals will be addressed as extra time allows.
